src/bridge/MemberListSyncer.ts

```diff
@@ -278,19 +277,28 @@ export class MemberListSyncer {
         // map the filtered rooms to a list of users to join
         // [Room:{reals:[uid,uid]}, ...] => [{uid,roomid}, ...]
         const entries: { roomId: string; displayName: string; userId: string; frontier: boolean}[] = [];
-        filteredRooms.forEach((roomInfo) => {
-            roomInfo.realJoinedUsers.forEach((uid, index) => {
+        const idleRegex = this.server.ignoreIdleUsersOnStartupExcludeRegex;
+        for (const roomInfo of filteredRooms) {
+            for (const uid of roomInfo.realJoinedUsers) {
+                if (this.server.ignoreIdleUsersOnStartup) {
+                    const idle = await this.ircBridge.activityTracker?.isUserOnline(
+                        uid, this.server.ignoreIdleUsersOnStartupAfterMs, false
+                    );
+                    if (!(idle?.online) && !idleRegex?.exec(uid)) {
+                        continue;
+                    }
+                }
                 entries.push({
                     roomId: roomInfo.id,
                     displayName: roomInfo.displayNames[uid],
                     userId: uid,
                     // Mark the first real matrix user f.e room so we can inject
                     // them first to get back up and running more quickly when there
                     // is no bot.
-                    frontier: (index === 0)
+                    frontier: !entries.find((ent) => ent.roomId === roomInfo.id),
```

Comment:
I don't understand this change in relation to the PR description or commits. Is this supposed to be here and it's related?

Reply:
Yep, it's important.

We previously assumed all joined members of a room would be joined to IRC so the first user would be considered the "frontier" user. However, since we *may* not bridge the first user, we need to do a slightly different check.